src/metabase/util/string.clj

```diff
@@ -0,0 +1,28 @@
+(ns metabase.util.string
+  "Util for building strings"
+  (:require
+   [metabase.util.i18n :refer [deferred-tru]]))
+
+
+(defn build-sentence
+  "Join parts of a sentence together to build a compound one.
+
+  Options:
+  - stop? (default true): whether to add a period at the end of the sentence
+
+  Examples:
+
+    (build-sentence [\"foo\" \"bar\" \"baz\"]) => \"foo, bar and baz.\"
+
+    (build-sentence [\"foo\" \"bar\" \"baz\"] :stop? false) => \"foo, bar and baz\"
+  "
+  ([parts]
+   (build-sentence parts :stop? true))
+  ([parts & {:keys [stop?]
+             :or   {stop? true}
+             :as _options}]
+   (when (seq parts)
+     (cond
+       (= (count parts) 1) (str (first parts) (when stop? \.))
+       (= (count parts) 2) (str (first parts) " " (deferred-tru "and")  " " (second parts) (when stop? \.))
+       :else               (str (first parts) ", " (build-sentence (rest parts)))))))
```

Comment:
This doesn't work for all languages, especially right-to-left ones. And I'm pretty sure not all languages will use commas and "and" in that way to create lists. I'm not sure what the solution is, but this isn't it. Maybe someone has attempted to solve the problem before and has some clues, especially on the FE

Reply:
https://shopify.engineering/internationalization-i18n-best-practices-front-end-developers

I read this the other day and I don't think they have a solution for dynamically building sentences.

Also,
1. Solving the generic problem of pluralization shouldn't be the goal of this PR
2. We only support 29 languages, 2 of which are right-to-left, so I think our pluralization is far from perfect. not that it's an excuse, but it supports the 1st point.
3. As this is exacted to be its own function, I think once we decided to solve this problem, we can solve it once for all.
4. Started a discussion here https://metaboat.slack.com/archives/C010L1Z4F9S/p1692241389613209

Reply:
Good points. Should we add a small FIXME at least explaining that this doesn't work for a lot of languages?

Reply:
I added that as a note in the function https://github.com/metabase/metabase/blob/ngoc-actions-error-handling-cleanup/src/metabase/util/string.clj